lib/matplotlib/tests/test_colorbar.py

```diff
@@ -979,6 +979,37 @@ def test_colorbar_extend_drawedges():
             np.testing.assert_array_equal(cbar.dividers.get_segments(), res)
 
 
+@image_comparison(['contourf_extend_patches.png'], remove_text=True,
+                  style='mpl20')
+def test_colorbar_contourf_extend_patches():
+    params = [
+        ('both', 5, ['\\', '//']),
+        ('min', 7, ['+']),
+        ('max', 2, ['|', '-', '/', '\\', '//']),
+        ('neither', 10, ['//', '\\', '||']),
+    ]
+
+    plt.rcParams['axes.linewidth'] = 2
+
+    fig = plt.figure(figsize=(10, 4))
+    subfigs = fig.subfigures(1, 2)
+
+    x = np.linspace(-2, 3, 50).reshape(1, -1)
+    y = np.linspace(-2, 3, 30).reshape(-1, 1)
+    z = np.cos(x) + np.sin(y)
+    x, y = x.flatten(), y.flatten()
```

Comment:
Less confusing to me than reshaping `x` and `y` twice:
```suggestion
    x = np.linspace(-2, 3, 50)
    y = np.linspace(-2, 3, 30)
    z = np.cos(x[np.newaxis, :]) + np.sin(y[:, np.newaxis])
```

Reply:
Fixed!